src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt

```diff
@@ -49,22 +50,23 @@ class MParticleAutopilot : Autopilot() {
         val token = airship.pushManager.pushToken
         MParticlePushProvider.instance.setRegistrationToken(token)
         airship.channel.addChannelListener(object : AirshipChannelListener {
-            override fun onChannelCreated(s: String) {
-                callChannelIdListener()
+            override fun onChannelCreated(channelId: String) {
+                
             }
 
-            override fun onChannelUpdated(s: String) {
-                callChannelIdListener()
+        })
+        fun callChannelIdListener() {
+            val channelIdListener = MParticle.getInstance()
+                ?.getKitInstance(MParticle.ServiceProviders.URBAN_AIRSHIP)
+            if (channelIdListener != null) {
+                (channelIdListener as ChannelIdListener).channelIdUpdated()
             }
+        }
+        //The onChange fun within the AirshipChannelListener was removed.
+        //PushNotificationStatusListener reporting changes instead (ref. [#https://github.com/urbanairship/android-library/blob/17.0.0/documentation/migration/migration-guide-16-17.md])
+        airship.pushManager.addNotificationStatusListener { callChannelIdListener() }
```

Comment:
It looks like all this is doing is setting the Airship channel ID as an attribute on mParticle. The channel in practice never changes but we do have a path in the SDK that we could regenerate it. If thats the case the `onCreate` will get called again.

I think all you need to do is sync on onAirshipReady and sync on channel created.